### PR TITLE
8 1 1 rebase main candidate

### LINKED ISSUE #478: Rebase Speedb on latest RocksDB release
latest release is RocksDB 8.1.1

### PR DESCRIPTION
**Pull request by Yuval-Ariel**
closes #478

### CODE REVIEW

File: db/db_impl/db_impl_compaction_flush.cc [hunk 1]
```diff
@@ -328,7 +329,7 @@ Status DBImpl::FlushMemTableToOutputFile(
         error_handler_.SetBGError(s, BackgroundErrorReason::kFlushNoWAL);
       }
     } else {
-      assert(s == log_io_s);
+      assert(s.code() == log_io_s.code() && s.subcode() == log_io_s.subcode());
```
Comment (ayulas) [inline db/db_impl/db_impl_compaction_flush.cc:332]: why not create a compare function like 
ASSERT_FULL_CODE(x,y) which does assert(x.code() == y.code() && x.subcode() == y.subcode()) ?
i see you have several places like that

**Review by ayulas — COMMENTED** [review]
i went over the changes files. i hope i didnt missed something . i have few comments

File: CMakeLists.txt [hunk 2]
```diff
@@ -1158,9 +1216,9 @@ if(ROCKSDB_BUILD_SHARED)
   else()
     set_target_properties(${ROCKSDB_SHARED_LIB} PROPERTIES
                           LINKER_LANGUAGE CXX
-                          VERSION ${rocksdb_VERSION}
-                          SOVERSION ${rocksdb_VERSION_MAJOR}
-                          OUTPUT_NAME "rocksdb${ARTIFACT_SUFFIX}")
+                          VERSION ${PROJECT_VERSION}
```
[the remark this answers is not in the document]
Comment (Yuval-Ariel) [reply CMakeLists.txt:1219]: speedb_VERSION and PROJECT_NAME are not interchangeable. But in any case, i dont think we should re-review the actual commits that were done in the past. just their rebase on 8.1.1

**Review by mrambacher — APPROVED** [review]
Please enter an issue for the tests that are disabled as part of this upgrade.

**Comment by Yuval-Ariel** [comment]
> Please enter an issue for the tests that are disabled as part of this upgrade.

there already is - 
https://github.com/speedb-io/speedb/issues/491
https://github.com/speedb-io/speedb/issues/492
https://github.com/speedb-io/speedb/issues/493
https://github.com/speedb-io/speedb/pull/499

**Comment by Yuval-Ariel** [comment]
done

**Pull request closed without merging**